kong/route_service_test.go

```diff
@@ -302,3 +310,51 @@ func TestRouteWithHeaders(T *testing.T) {
 	err = client.Routes.Delete(defaultCtx, createdRoute.ID)
 	assert.NoError(err)
 }
+
+func TestRoutesValidation(T *testing.T) {
```

Comment:
I think we should also add a test case for `traditional_compatible` (e.g. invalid regex).